scipy/signal/signaltools.py

```diff
@@ -428,7 +456,222 @@ def fftconvolve(in1, in2, mode="full"):
+    # catch when more precision required than float provides (representing a
+    # integer as float can lose precision in fftconvolve if larger than 2**52)
+    if any([_numeric_arrays([x], kinds='ui') for x in [volume, kernel]]):
+        max_value = int(np.abs(volume).max()) * int(np.abs(kernel).max())
+        max_value *= int(min(volume.size, kernel.size))
+        if max_value > 2**np.finfo('float').nmant - 1:
+            return False
+
+    if volume.dtype.kind not in 'buifc' and kernel.dtype.kind not in 'buifc':
+        return False
+
+    return True
+
+
+def _timeit_fast(stmt="pass", setup="pass", repeat=3):
+    """
+    Faster, less painstaking version of timeit_auto for automated 
+    testing of many inputs.
+
+    Will do only 1 loop (like IPython's timeit) with no repetitions 
+    (unlike IPython) for very slow functions.  Only does enough loops 
+    to take 5 ms, which seems to produce similar results on Windows at 
+    least, and avoids doing an extraneous cycle that isn't measured.
+    """
+    t = Timer(stmt, setup)
+
+    # determine number so that 5 ms <= total time
+    x = 0
+    for i in range(0, 10):
+        number = 10**i
+        x = t.timeit(number)  # seconds
+        if x >= 5e-3 / 10:  # 5 ms for final test, 1/10th that for this one
+            break
+    if x > 1:  # second
+        # If it's macroscopic, don't bother with repetitions
+        best = x
+    else:
+        number *= 10
+        r = t.repeat(repeat, number)
+        best = min(r)
+
+    usec = best * 1e6 / number
+    return usec
+
+def choose_conv_method(in1, in2, mode='full', measure=False):
+    """
+    A method to find the fastest convolution method.
+
+    Parameters
+    ----------
+    in1 : array_like
+        The first argument passed into the convolution function.
+    in2 : array_like
+        The second argument passed into the convolution function.
+    mode : str {'full', 'valid', 'same'}, optional
+        A string indicating the size of the output:
+
+        ``full``
+           The output is the full discrete linear convolution
+           of the inputs. (Default)
+        ``valid``
+           The output consists only of those elements that do not
+           rely on the zero-padding.
+        ``same``
+           The output is the same size as `in1`, centered
+           with respect to the 'full' output.
+    measure : bool or num, optional
+        If True, run and time the convolution with `in1` and `in2` and return
+        the fastest method. If False (default), predict the fastest method
+        using precomputed values.
+
+    Returns
+    -------
+    method : str
+        A string indicating which convolution method is fastest, either
+        'direct' or 'fft'
+    time_ratio : num, optional
+        This value is only returned if `measure` is not False and is how many
+        times longer the Fourier transform method of convolution took than the
+        direct method, ``time_fft / time_direct``.
```

Comment:
Hmm, I don't think that's as useful as returning both values for the case of `measure=True`. You might actually want to know how long it took.